ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/impl/QueryResultImplTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import static org.junit.Assert.assertThrows;
+
+import io.confluent.ksql.api.client.Row;
+import io.vertx.core.Context;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.reactivestreams.Subscriber;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryResultImplTest {
+
+  @Mock
+  private Context context;
+  @Mock
+  private Subscriber<Row> subscriber;
+
+  private QueryResultImpl queryResult;
+
+  @Before
+  public void setUp() {
+    queryResult = new QueryResultImpl(context, "queryId", Collections.emptyList(), Collections.emptyList());
+  }
+
+  @Test
+  public void shouldNotSubscribeIfPolling() {
```

Comment:
Imho I would like to see these kinds of tests conducted using the actual API and no mocks.

As you know I'm not a fan of fine grained unit tests and mocks as they can constrain the implementation, and often what you're testing doesn't really correspond to what the system really does thus resulting in bugs slipping through and a false sense of security.

Reply:
Sure, I'll add equivalent tests to `ClientTest` as part of revamping / improving test coverage. My vote is to leave these unit tests in place, though, until we see them become brittle. I think it's useful to be able to scan a test file and understand the key pieces of functionality for a class without having to dig through integration tests. Though I guess Java docs on the class/interface itself should serve this purpose in most cases, so maybe that's not a good reason...

Reply:
As long as there's equivalent test coverage on the actual API I think that's fine. And as long as you won't be upset if I end up deleting them after spending an hour trying to refactor them if the implementation changes ;)

Reply:
Update: going to add these additional tests in a follow-up PR.

> And as long as you won't be upset if I end up deleting them after spending an hour trying to refactor them if the implementation changes ;)

Fine by me ;)